.github/setup_python/action.yml

```diff
@@ -1,27 +1,45 @@
 name: "Setup Python"
 description: "Sets up Python for use in actions with caching and copy localsettings"
 
+inputs:
+  requirements-file:
+    description: "name of the requirements file to install"
+    required: false
+    default: requirements-dev.txt
+
 runs:
   using: "composite"
   steps:
     - name: Setup python
+      id: python
       uses: actions/setup-python@v4
       with:
-        python-version: '3.10'
+        python-version: "3.10"
 
     - name: Restore venv cache
-      uses: syphar/restore-virtualenv@v1
       id: cache-virtualenv
-
-    - name: Restore pip download cache
-      uses: syphar/restore-pip-download-cache@v1
-      if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+      uses: actions/cache@v4
+      with:
+        path: .venv
+        key: venv-${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{ hashFiles( 'requirements*.txt' ) }}
 
     - name: Install dependencies
-      run: pip install -r requirements-dev.txt
+      run: |
+        python -m venv .venv
+        source .venv/bin/activate
+        pip install -r "${{ inputs.requirements-file }}"
       shell: bash
       if: steps.cache-virtualenv.outputs.cache-hit != 'true'
```

Comment:
Not sure if this is a problem or not, but does this mean that the cache key is the same regardless of which requirements file is passed via inputs.requirements-file? If I am not mistaken, we probably want both the hash of all files, but still the name of the file that is used for pip in the following step. Otherwise, we might use a cache for requirements-Dev when we really only wanted requirements.txt - or is that prevented elsewhere?

Reply:
you are right, I forgot this when reverting to the requirements-file input. Should now be fixed

Reply:
The current version has the problem from my first comment again. The resulting venv directory generated by pip is a function of
- the OS,
- the python version,
- what the contents of _all_ requirements files are (because they include each other), and
- which requirements file is actually used.

So I would think that we would use something like `venv-${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{ hashFiles( 'requirements*.txt' ) }}-${{ inputs.requirements-file }}`

Reply:
Ah well, I did not understand the initial problem... Your line works and fixes the problem :+1:  
As a drawback, the "requirements.txt" cache is cleared when only -dev dependencies change. I think this should be fine however.